File: JavaScript_Basic_04_2023_SolutionsOfProblems/05.Ex_PB-JS-While-Loop/P03.Vacation.js
function vacation(input) {
    let tripCost = Number(input[0]);
    let savedMoney = Number(input[1]);
    let i = 2, spendDayCounter = 0;
    while (spendDayCounter < 5 && savedMoney < tripCost) {
        let action = input[i];
        i++;
        if (action === "spend") {
            spendDayCounter++;
            savedMoney -= Number(input[i]);
            if (savedMoney < 0) {
                savedMoney = 0;
            }
        } else {
            spendDayCounter = 0;
            savedMoney += Number(input[i]);
        }
        i++;
    }
    if (spendDayCounter === 5) {
        console.log("You can't save the money.");
        console.log((i - 2) / 2);
    } else {
        console.log(`You saved the money for ${(i - 2) / 2} days.`);
    }
}
vacation(["110",
    "60",
    "spend",
    "10",
    "spend",
    "10",
    "spend",
    "10",
    "spend",
    "10",
    "spend",
    "10"])

File: SoftUni_JAVA_BASIC_Jan_2023_SolutionsOfProblems/P05Ex_While_Loop/P03_Vacation.java
package P05Ex_While_Loop;

import java.util.Scanner;

public class P03_Vacation {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        double priceTrip = Double.parseDouble(scanner.nextLine());
        double savedMoney = Double.parseDouble(scanner.nextLine());

        int daysSpend = 0, daysTotal = 0;
        double money = 0;
        String action = "";

        while (savedMoney < priceTrip && daysSpend < 5) {
            action = scanner.nextLine();
            money = Double.parseDouble(scanner.nextLine());
            daysTotal++;
            if (action.equals("spend")) {
                daysSpend++;
                savedMoney -= money;
                if (savedMoney < 0) {
                    savedMoney = 0;
                }
            } else {
                daysSpend = 0;
                savedMoney += money;
            }
        }
        if (daysSpend == 5) {
            System.out.printf("You can't save the money.%n%d", daysTotal);
        } else {
            System.out.printf("You saved the money for %d days.", daysTotal);
        }
    }
}
